File: ReportChecker/src/app/entities/comment-entity.ts
import {Moment} from 'moment';
import {PatchEntity} from './patch-entity';

export interface CommentEntity {
  id: string;
  userId: string;
  content: string | null;
  status: string | null;
  progressStatus: string | null;
  isRead: boolean | null;
  patch?: PatchEntity;
  createdAt: Moment;
  updatedAt: Moment | null;
}

File: ReportChecker.Api/ReportChecker.DataAccess/Entities/CommentEntity.cs
using System.ComponentModel.DataAnnotations;
using ReportChecker.Models;

namespace ReportChecker.DataAccess.Entities;

public class CommentEntity
{
    public Guid CommentId { get; init; }
    public Guid IssueId { get; init; }
    public Guid UserId { get; init; }
    [MaxLength(8000)] public string? Content { get; init; }
    public IssueStatus? Status { get; init; }
    public ProgressStatus? ProgressStatus { get; init; }
    public DateTime CreatedAt { get; init; }
    public DateTime? ModifiedAt { get; init; }
    public DateTime? DeletedAt { get; init; }

    public IssueEntity Issue { get; init; } = null!;
}